plugins/builtin/apm/nomad/plugin/metrics.go

```diff
@@ -91,3 +95,47 @@ func (a *APMPlugin) Query(q string) (float64, error) {
 
 	return result, nil
 }
+
+func parseQuery(q string) (*Query, error) {
+	mainParts := strings.SplitN(q, "/", 3)
+	if len(mainParts) != 3 {
+		return nil, fmt.Errorf("expected <query>/<job>/group>, received %s", q)
+	}
+
+	query := &Query{
+		Group: mainParts[1],
+		Job:   mainParts[2],
+	}
+
+	opMetricParts := strings.SplitN(mainParts[0], "_", 2)
+	if len(opMetricParts) != 2 {
+		return nil, fmt.Errorf(`expected <operation>_<metric>, received "%s"`, mainParts[0])
+	}
+
+	op := opMetricParts[0]
+	metric := opMetricParts[1]
+
+	switch metric {
+	case "cpu":
+		query.Metric = "nomad.client.allocs.cpu.total_percent"
+	case "memory":
+		query.Metric = "nomad.client.allocs.memory.usage"
+	default:
+		query.Metric = metric
+	}
+
+	switch op {
+	case "sum":
+		fallthrough
+	case "avg":
+		fallthrough
+	case "min":
+		fallthrough
+	case "max":
+		query.Operation = op
```

Comment:
for me this is a bit easier to read and more concise

```suggestion
	case "sum", "avg", "min", "max":
		query.Operation = op
```